rlog.go

```diff
@@ -317,13 +445,30 @@ func Initialize(config rlogConfig) {
 	}
 
 	// ... but if requested we'll also create and/or append to a logfile
-	if config.logFile == "" {
-		logWriterFile = nil
-	} else {
-		newLogFile, err := os.OpenFile(config.logFile,
-			os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
-		if err == nil {
-			logWriterFile = log.New(newLogFile, "", 0)
+	var newLogFile *os.File
+	if currentLogFileName != config.logFile { // something changed
+		if config.logFile == "" {
+			// no more log output to a file
+			logWriterFile = nil
+		} else {
+			// Check if the logfile was changed or was set for the first
+			// time. Only then do we need to open/create a new file.
+			// We also do this if for some reason we don't have a log writer
+			// yet.
+			if currentLogFileName != config.logFile || logWriterFile == nil {
+				newLogFile, err = os.OpenFile(config.logFile,
+					os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+				if err == nil {
```

Comment:
Then it should continue to write to the old log writer. Which won't exactly work here either, because of other issues. Need to fix that.